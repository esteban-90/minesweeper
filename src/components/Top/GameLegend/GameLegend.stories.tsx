import type { Story, Meta } from '@storybook/react'
import { GameLegend, type GameLegendProps } from './GameLegend'

export default {
  title: 'Components/Top/Legend',
  component: GameLegend,
} as Meta

const Template: Story<GameLegendProps> = (args) => <GameLegend {...args} />

export const GameLegendExample = Template.bind({})

GameLegendExample.args = {
  feature: 'flag',
  firstAction: 'ctrl',
  secondAction: 'click',
}
