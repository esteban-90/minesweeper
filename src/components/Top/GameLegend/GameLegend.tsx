import type { FC } from 'react'
import styled from '@emotion/styled'

const Parent = styled.legend`
  font-size: 1em;
  margin: 0 auto 2vw;
  line-height: 1.25em;
`

const Combo = styled.code`
  background-color: #e3e3e3;
`

const FirstAction = styled.span`
  color: #ec433c;
`

const SecondAction = styled.span`
  color: #2a48ec;
`

export interface GameLegendProps {
  /**
   * Feature that should be activated after first + second actions
   */
  feature: string
  /**
   * First action
   */
  firstAction: string
  /**
   * Second action
   */
  secondAction: string
}

export const GameLegend: FC<GameLegendProps> = ({ feature, firstAction, secondAction }): JSX.Element => {
  return (
    <Parent>
      <strong>{feature}:</strong>
      <Combo>
        <FirstAction>{firstAction}</FirstAction> + <SecondAction>{secondAction}</SecondAction>
      </Combo>
    </Parent>
  )
}
