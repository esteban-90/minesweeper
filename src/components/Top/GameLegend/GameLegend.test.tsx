import { render } from '@testing-library/react'
import { GameLegend } from './GameLegend'

describe('GameLegend test cases:', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<GameLegend feature='Flag' firstAction='ctrl' secondAction='click' />)
    expect(asFragment()).toMatchSnapshot()
  })
})
