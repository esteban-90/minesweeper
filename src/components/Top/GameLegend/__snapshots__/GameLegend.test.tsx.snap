// Vitest Snapshot v1

exports[`GameLegend test cases: > should render correctly 1`] = `
<DocumentFragment>
  <legend
    class="css-17qz6a2"
  >
    <strong>
      Flag:
    </strong>
    <code
      class="css-h9xasx"
    >
      <span
        class="css-udfepp"
      >
        ctrl
      </span>
       + 
      <span
        class="css-9c23k1"
      >
        click
      </span>
    </code>
  </legend>
</DocumentFragment>
`;
